src/exo/API_scheduling.py

```diff
@@ -1001,14 +1002,46 @@ def bind_expr(proc, expr_cursors, new_name):
 # Sub-procedure Operations
 
 
-@sched_op([NameA, StmtCursorA, DictA])
-def extract_subproc(proc, subproc_name, body_stmt, order=dict()):
-    """
-    Documentation
+@sched_op([NameA, BlockCursorA, DictA])
+def extract_subproc(proc, subproc_name, block, order=dict()):
     """
-    stmt = body_stmt._impl
-    passobj = scheduling.DoExtractMethod(proc, subproc_name, stmt, order)
-    return passobj.result(), passobj.subproc()
+    Extract a block as a subprocedure with the name `subproc_name`.
+
+    # TODO: deprecate `order` and swap the order of `subproc_name` and `block`.
```

Comment:
Once we confirm the CI passes and we review this code, I plan to do this which will break the BLAS code.

Reply:
Why do you want to deprecate `order`? There will be no way of permuting the argument ordering otherwise.

Reply:
I'm mildly against deprecating order

Reply:
It is not a valuable parameter if the implementation is written properly. 

Previously, it was being used as a workaround because the parameters of the subproc would be created in a different order than the parent proc. Now, that they follow the same order. There is no good reason for it. If you notice, currently, order is not being used in the implementation and all the tests pass.